tcbuilder/backend/build.py

```diff
@@ -242,18 +281,11 @@ def parse_config_file(config_path, schema_path=DEFAULT_SCHEMA_FILE, substs=None)
         raise PathNotExistError(
             f"Build configuration file '{config_path}' does not exist.")
 
-    # Load the YAML configuration file (user-supplied):
-    with open(config_path) as file:
-        try:
-            config = yaml.load(file, Loader=yaml.FullLoader)
 
-        except yaml.YAMLError as ex:
-            error_exc = ParseError(getattr(ex, "problem", "parsing error"))
-            error_exc.set_source(file=config_path)
-            if hasattr(ex, "problem_mark"):
-                mark = getattr(ex, "problem_mark")
-                error_exc.set_source(line=mark.line, column=mark.column)
-            raise error_exc
+    if config_path.endswith(".json"):
+        config = _load_config_file_json(config_path)
+    elif config_path.endswith(".yaml") or config_path.endswith(".yml"):
+        config = _load_config_file_yaml(config_path)
```

Comment:
This should probably provide a clean error message (and abort) if no supported extension was found. Now I think it will raise an exception for `config` being undefined.